pkg/providers/nutanix/template.go

```diff
@@ -161,8 +161,17 @@ func buildTemplateMapCP(
 	format := "cloud-config"
 	apiServerExtraArgs := clusterapi.OIDCToExtraArgs(clusterSpec.OIDCConfig).
 		Append(clusterapi.AwsIamAuthExtraArgs(clusterSpec.AWSIamConfig)).
-		Append(clusterapi.PodIAMAuthExtraArgs(clusterSpec.Cluster.Spec.PodIAMConfig)).
+		Append(clusterapi.APIServerExtraArgs(clusterSpec.Cluster.Spec.ControlPlaneConfiguration.APIServerExtraArgs)).
 		Append(clusterapi.EtcdEncryptionExtraArgs(clusterSpec.Cluster.Spec.EtcdEncryption))
+	for k, v := range clusterapi.PodIAMAuthExtraArgs(clusterSpec.Cluster.Spec.PodIAMConfig) {
```

Comment:
This for loop can be extracted out to a separate file right? How much scope is it to leverage the configure method we use for snow I think?

Reply:
Created new function in `extraargs.go` which runs this for loop and sets the `podIamAuthExtraArgs`